pkg/api/source/grpc_adapter.go

```diff
@@ -23,13 +25,50 @@ type (
 	StreamInput struct {
 		// Configs is a list of Source configurations specified by users.
 		Configs []*Config
+		// Context holds streaming context.
+		Context StreamInputContext
+	}
+
+	// StreamInputContext holds streaming context.
+	StreamInputContext struct {
+		// IsInteractivitySupported is set to true only if communication platform supports interactive Messages.
+		IsInteractivitySupported bool
+
+		// KubeConfig is the path to kubectl configuration file.
+		KubeConfig string
+
+		// ClusterName is the name of underlying Kubernetes cluster which is provided by end user.
+		ClusterName string
 	}
 
 	// StreamOutput holds the output of the Stream function.
 	StreamOutput struct {
 		// Output represents the streamed events. It is from start of plugin execution.
+		// Deprecated: Use the Message field instead.
+		//
+		// Migration path:
+		//
+		//	Old approach:
+		//	  return executor.StreamOutput{
+		//	  	Output: out,
+		//	  }
+		//
+		//	New approach:
+		//	  return executor.StreamOutput{
+		//	  	Message: api.NewPlaintextMessage(out, true),
+		//	  }
 		Output chan []byte
-		// TODO: we should consider adding error feedback channel too.
+		// Message represents the streamed events with metadata. It is from start of plugin consumption.
+		// You can construct a complex message.data or just use one of our helper functions:
+		//   - api.NewCodeBlockMessage("body", true)
+		//   - api.NewPlaintextMessage("body", true)
+		Message chan Message
+	}
+
+	Message struct {
+		Data      api.Message
+		Metadata  any
+		Telemetry map[string]interface{}
 	}
```

Comment:
```go
	StreamOutput struct {
		Event chan Event
	}

	Event struct {
		Message          api.Message
		Metadata         any
		AnonymizedFields map[string]interface{}
	}
```

WDYT about such terminology? @huseyinbabal @pkosiec ?

Reply:
I am ok with the wording, changing accordingly